concepts/identity.md

```diff
@@ -0,0 +1,283 @@
+
+This ownership CAN be transferred to a different Ethereum address. Activation of this feature SHOULD be optional, and MUST include a user-modifiable grace period during which reverting to previous ownership is possible.
+
+A Meta Identity owner CAN perform transfer of ownership. A Meta Identity owner CAN appoint a number of Ethereum addresses to act as *custodians*. These CAN through a vote perform transfer of ownership. A vote MUST require a message signed by a respective private key. The *minimum number of addresses* that must make up a custodian role, and *minimum ratio of votes* required to execute a transfer CAN ONLY be determined by the Meta Identity owner.
+
+### Implementation
+
+
+The Meta Identity uses ENS as backend for storage and retrieval, interfaced by the Identity and IdentityResolver contracts.
+
+The Identity contract is upgradeable and MUST be registered in the IdentityController contract. As the IdentityController instantiates all permanent storage, and guarantees that the contents of these cannot be altered between upgrades. Therefore only the instance of the Identity contract that has been registered with the IdentityController can be considered valid for use.
+
+In the following we assume the META-ID to be `sha3("foo")`.
+
+#### Add a new identity
+
+```
+Identity.register(bytes32 metaidhash, address owner);
+```
+
+This will make `owner` the owner of the META-ID, along with the ENS node `namehash("foo.id.meta")`. If the META-ID has already been registered, the method will throw an exception.
+
+#### Add metadata for identity
+
+```
+IdentityResolver.setContent(bytes32 node, bytes32 hash);
+```
+
+sets the content hash `hash` as the metadata for the node `namehash("foo.id.meta")`
+
+#### Retrieve metadata for identity
+
+```
+IdentityResolver.content(bytes32 node);
+```
+
+gets the content hash of the metadata for the node namehash("foo.id.meta")
+
+#### Retrieve the current valid Identity contract
+
+```
+IdentityController.current();
+```
+
+gets the address of the current valid Identity Contract
+
+#### Not yet implemented
+
+* Identity Custody Scheme contracts
+* events for invalid attempts of changes to identity
+* events for initiated change identity owner vote
+* events for notifying identity monitors to request refreshing their indexing (used for updating public keys) ...
+
+### Metadata
+
+Meta Identity metadata MUST constitute two parts; Core and Auxiliary. Core metadata is REQUIRED and can be defined only once. The Auxiliary metadata is OPTIONAL, and an arbitrary number of Auxiliary metadata items may be defined.
+
+Metadata may ONLY be changed by the Meta Identity owner.
+
+#### Core
+
+The following properties are REQUIRED:
+
+```
+id		: META-ID // reverse point to META-ID this metadata belongs to
+name		: string // name whom identity represents
+parent		: CID // pointer to previous version of this metadata
+```
+#### Auxiliary
+
+Auxiliary metadata items MUST be grouped in two subfolders; *key* and *aux*.
```

Comment:
What is a "subfolder"? Are we talking about a filesystem here?

Reply:
This alludes to a Swarm manifest representation of the metadata. Maybe we should change this to be specific to the META Object representation?

Reply:
Hmm I mean Swarm manifests don't have "folders" either (I think they [actually should do though](https://github.com/ethereum/go-ethereum/issues/14349) :wink:).

Perhaps something like:

```
The auxiliary META object must have "key" and "aux" properties.
```

Reply:
It seems we are reverting back to just `aux` re: clarification from @lmars to question from @zelig in the contract PR?

Reply:
Yes, seems that way. Do we need to discuss this or are we all agreed? I still think the separation of "key" and "aux" makes sense as they are quite distinct entities but am happy to go with the majority.

Reply:
I vote let's keep it as simple as possible. `aux` only